### PR TITLE
[21.05] gd: 2.3.0 -> 2.3.2, add patch for partial CVE-2021-40812 fix

### PR DESCRIPTION
**Pull request by risicle**
###### Motivation for this change
Backport of #138981 without the reversion as that was already done on this branch in #138851

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
@ofborg build php80Extensions.gd

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 139225` at f303ebf5 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>142 packages marked as broken and skipped:</summary>
  <ul>
    <li>adoptopenjdk-hotspot-bin-13</li>
    <li>adoptopenjdk-hotspot-bin-14</li>
    <li>adoptopenjdk-jre-hotspot-bin-13</li>
    <li>adoptopenjdk-jre-hotspot-bin-14</li>
    <li>airtame</li>
    <li>asymptote</li>
    <li>attract-mode</li>
    <li>autotrace</li>
    <li>betaflight-configurator</li>
    <li>blastem</li>
    <li>bless</li>
    <li>cassandra_2_1</li>
    <li>cassandra_2_2</li>
    <li>colmapWithCuda</li>
    <li>cura_stable</li>
    <li>deeptools</li>
    <li>devede</li>
    <li>displaycal</li>
    <li>dockbarx</li>
    <li>dvd-slideshow</li>
    <li>electron_3</li>
    <li>electron_4</li>
    <li>electron_5</li>
    <li>exodus</li>
    <li>ffmpeg_2</li>
    <li>ffmpeg_2_8</li>
    <li>ffmpeg_3</li>
    <li>ffmpeg_3_4</li>
    <li>flamerobin</li>
    <li>forge</li>
    <li>gmtk</li>
    <li>gnome_mplayer</li>
    <li>gnuradio3_7</li>
    <li>gnuradio3_7Packages.ais</li>
    <li>gnuradio3_7Packages.gnuradio</li>
    <li>gnuradio3_7Packages.gsm</li>
    <li>gnuradio3_7Packages.limesdr</li>
    <li>gnuradio3_7Packages.nacl</li>
    <li>gnuradio3_7Packages.osmosdr</li>
    <li>gnuradio3_7Packages.rds</li>
    <li>gopro</li>
    <li>grass</li>
    <li>gtkd</li>
    <li>hplipWithPlugin_3_18_5</li>
    <li>hplip_3_18_5</li>
    <li>hydrus</li>
    <li>impressive</li>
    <li>intel-compute-runtime</li>
    <li>intel-media-driver</li>
    <li>invoice2data</li>
    <li>isabelle</li>
    <li>jboss</li>
    <li>keyfinder</li>
    <li>libav</li>
    <li>libav_0_8</li>
    <li>libav_11</li>
    <li>libav_12</li>
    <li>libgroove</li>
    <li>linuxPackages_hardkernel_4_14.akvcam</li>
    <li>linuxPackages_hardkernel_latest.akvcam</li>
    <li>linuxPackages_testing_bcachefs.akvcam</li>
    <li>loxodo</li>
    <li>luxcorerender</li>
    <li>mame</li>
    <li>metamorphose2</li>
    <li>natron</li>
    <li>obs-ndi</li>
    <li>odpdown</li>
    <li>openems</li>
    <li>openrw</li>
    <li>opentsdb</li>
    <li>pcsxr</li>
    <li>pdfdiff</li>
    <li>photoflow</li>
    <li>pig</li>
    <li>plover.stable</li>
    <li>poedit</li>
    <li>polymake</li>
    <li>processing</li>
    <li>pulseeffects-legacy</li>
    <li>pulseeffects-pw</li>
    <li>python38Packages.HTSeq</li>
    <li>python38Packages.caffe</li>
    <li>python38Packages.cnvkit</li>
    <li>python38Packages.dask-xgboost</li>
    <li>python38Packages.datashader</li>
    <li>python38Packages.fipy</li>
    <li>python38Packages.foundationdb51</li>
    <li>python38Packages.foundationdb52</li>
    <li>python38Packages.foundationdb60</li>
    <li>python38Packages.foundationdb61</li>
    <li>python38Packages.foxdot</li>
    <li>python38Packages.ignite</li>
    <li>python38Packages.python-csxcad</li>
    <li>python38Packages.python-openems</li>
    <li>python38Packages.streamz</li>
    <li>python38Packages.tensorflowWithCuda</li>
    <li>python38Packages.textacy</li>
    <li>python39Packages.HTSeq</li>
    <li>python39Packages.caffe</li>
    <li>python39Packages.cnvkit</li>
    <li>python39Packages.dask-xgboost</li>
    <li>python39Packages.fipy</li>
    <li>python39Packages.foundationdb51</li>
    <li>python39Packages.foundationdb52</li>
    <li>python39Packages.foundationdb60</li>
    <li>python39Packages.foundationdb61</li>
    <li>python39Packages.foxdot</li>
    <li>python39Packages.python-csxcad</li>
    <li>python39Packages.python-openems</li>
    <li>python39Packages.streamz</li>
    <li>python39Packages.tensorflowWithCuda</li>
    <li>qemu_xen</li>
    <li>qemu_xen-light</li>
    <li>qemu_xen_4_10</li>
    <li>qemu_xen_4_10-light</li>
    <li>qlandkartegt</li>
    <li>ring-daemon</li>
    <li>rocm-opencl-icd</li>
    <li>rocm-opencl-runtime</li>
    <li>sonic-pi</li>
    <li>spark</li>
    <li>spotifywm</li>
    <li>spring</li>
    <li>springLobby</li>
    <li>stepmania</li>
    <li>termplay</li>
    <li>tilix</li>
    <li>tlaps</li>
    <li>torchat</li>
    <li>tvheadend</li>
    <li>twinkle</li>
    <li>unicap</li>
    <li>vbam</li>
    <li>vcs</li>
    <li>winpdb</li>
    <li>wt</li>
    <li>wt3</li>
    <li>wt4</li>
    <li>xfce.xfce4-dockbarx-plugin</li>
    <li>xfce.xfce4_dockbarx_plugin</li>
    <li>xpdf</li>
  </ul>
</details>
<details>
  <summary>4750 packages skipped due to time constraints:</summary>
  <ul>
    <li>AusweisApp2</li>
    <li>DisnixWebService</li>
    <li>MIDIVisualizer</li>
    <li>R</li>
    <li>SDL2_gfx</li>
    <li>SDL2_image</li>
    <li>SDL2_mixer</li>
    <li>SDL2_net</li>
    <li>SDL2_ttf</li>
    <li>SDL_gpu</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>33 packages built successfully:</summary>
  <ul>
    <li>SDL2</li>
    <li>openjdk16-bootstrap (adoptopenjdk-hotspot-bin-15)</li>
    <li>openjdk8-bootstrap (adoptopenjdk-hotspot-bin-8)</li>
    <li>chromaprint</li>
    <li>ffmpeg (ffmpeg_4)</li>
    <li>gd</li>
    <li>geoclue2</li>
    <li>glib-networking (gnome.glib-networking ,gnome.glib_networking)</li>
    <li>gnome.gnome-desktop (gnome.gnome_desktop)</li>
    <li>gtk3 (gnome.gtk ,gtk3-x11)</li>
    <li>gssdp</li>
    <li>gst_all_1.gst-libav</li>
    <li>gst_all_1.gst-plugins-base</li>
    <li>gst_all_1.gst-plugins-ugly</li>
    <li>gtkspell3</li>
    <li>gupnp</li>
    <li>gupnp-igd</li>
    <li>lcov</li>
    <li>libnice</li>
    <li>libproxy</li>
    <li>libva</li>
    <li>libva-minimal</li>
    <li>libxkbcommon (libxkbcommon_8)</li>
    <li>networkmanager</li>
    <li>openconnect (openconnect_gnutls)</li>
    <li>pantheon.elementary-icon-theme</li>
    <li>pantheon.granite</li>
    <li>perl532Packages.GD</li>
    <li>qrencode</li>
    <li>stoken</li>
    <li>vulkan-loader</li>
    <li>wayland-protocols</li>
    <li>wayland-scanner</li>
  </ul>
</details>

**Comment by Artturin** [comment]
merge?

**Comment by risicle** [comment]
If someone approves it, sure.

There are quite a few unreviewed & therefore unmerged PRs under the security label. Not all of them mine.

**Review by Artturin — APPROVED** [review]
Result of `nixpkgs-review pr 139225` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wayland</li>
  </ul>
</details>
